### PR TITLE
fix: add missing trait impls for integer types to stdlib

### LINKED ISSUE #5733: No impl Eq for u16
### Aim

```rs
fn main() {
    let one: u16 = 1;
    let _ = (one, 1) == (1, 1);
}
```

### Expected Behavior

The program to execute successfully

### Bug

The `cmp` traits don't have implementations for `u16` or  `i16` since it was added later.

### To Reproduce

1.
2.
3.
4.

### Workaround

None

### Workaround Description

_No response_

### Additional Context

_No response_

### Project Impact

None

### Blocker Context

_No response_

### Nargo Version

_No response_

### NoirJS Version

_No response_

### Proving Backend Tooling & Version

_No response_

### Would you like to submit a PR for this Issue?

None

### Support Needs

_No response_

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

## Problem\*

Resolves #5733 

## Summary\*

This PR implements some missing trait implementations for numeric types. This covers `u16` as mentioned in the issue but also `u1`s

## Documentation\*

Check one:
- [x] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[For Experimental Features]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Review by jfecher — COMMENTED** [review]
LGTM but can you add impls for Default and Hash as well?

**Comment by TomAFrench** [comment]
Ah I didn't see those. Thanks, fixed now.

File: noir_stdlib/src/default.nr [hunk 1]
```diff
@@ -17,7 +17,9 @@ comptime fn derive_default(s: StructDefinition) -> Quoted {
 
 impl Default for Field { fn default() -> Field { 0 } }
 
+impl Default for u1 { fn default() -> u1 { 0 } }
 impl Default for u8 { fn default() -> u8 { 0 } }
+impl Default for u16 { fn default() -> u16 { 0 } }
```
Comment (jfecher) [inline noir_stdlib/src/default.nr:22]: Missing `impl Default for i16`

File: noir_stdlib/src/hash/mod.nr [hunk 2]
```diff
@@ -195,12 +195,24 @@ impl Hash for Field {
...
+    }
+}
+
 impl Hash for u8 {
     fn hash<H>(self, state: &mut H) where H: Hasher{
         H::write(state, self as Field);
     }
 }
 
+impl Hash for u16 {
```
Comment (jfecher) [inline noir_stdlib/src/hash/mod.nr:210]: Missing `impl Hash for i16`

**Review by jfecher — APPROVED** [review]

**Pull request merged**